plugins/ltac2/tac2tactics.ml

```diff
@@ -405,6 +405,28 @@ let typeclasses_eauto strategy depth dbs =
 
 let unify x y = Tactics.unify x y
 
+(** Transparent state *)
+
+let current_ts () =
+  Proofview.Goal.enter_one @@ fun gl ->
```

Comment:
I'm not sure this is the right thing here. In particular this will fail if there is more than one goal.

Alternatively, we could use the global environment, since (after a discussion with @Janno), there does not seem to be a way to change the transparency state of a single goal.

Reply:
If by global environment you mean Proofview.tclENV then yes do that